### PR TITLE
 Fix leaking of GroupStore listeners in RoomList 

### PR DESCRIPTION
**Pull request by lukebarnard1**
Built on https://github.com/matrix-org/matrix-react-sdk/pull/1647

### CODE REVIEW

File: src/stores/GroupStore.js [hunk 1]
```diff
@@ -110,6 +110,12 @@ export default class GroupStore extends EventEmitter {
         this._fetchSummary();
         this._fetchRooms();
         this._fetchMembers();
+
+        return {
```
Comment (ara4n) [inline src/stores/GroupStore.js]: i'm a bit surprised that one can return a object like this from registerListener - is this just ignorance, or can we have a comment to explain what API this comes from?

Comment (ara4n) [reply src/stores/GroupStore.js]: or is this adhoc of our own devising? in which case, it really should be jsdoc'd on the method as the return type

**Comment by ara4n** [comment]
lgtm modulo comment req

File: src/stores/GroupStore.js (hunk 1 above)
Comment (lukebarnard1) [reply src/stores/GroupStore.js]: This is fairly similar to the `flux/utils.Store` way of doing things, which returns `{ stop: () => {...} }`

Comment (ara4n) [reply src/stores/GroupStore.js]: ok. jsdoc would be good enough, imo: "This method returns an object which callers can use to invoke an unregister hook to..." or whatever.

Comment (lukebarnard1) [reply src/stores/GroupStore.js]: Awesome, I did both in the end 👍

**Pull request merged**